src/formatDate.js

```diff
@@ -48,9 +48,40 @@
  *
  * @returns {string}
  */
-
 function formatDate(date, fromFormat, toFormat) {
-  // write code here
+  const inputDate = {};
+  const orderedArr = [];
+
+  for (let i = 0; i < fromFormat.length - 1; i++) {
+    const type = fromFormat[i][0].toLocaleLowerCase();
+    const value = date.split(fromFormat[3])[i];
+
+    inputDate[type] = value;
+
+    if (type === 'y') {
+      inputDate.yearFormat = fromFormat[i].length;
+    }
+  }
+
+  for (let i = 0; i < toFormat.length - 1; i++) {
+    const type = toFormat[i][0].toLocaleLowerCase();
+
+    if (type === 'y') {
+      const toYearFormat = toFormat[i].length;
+
+      if (toYearFormat < inputDate.yearFormat) {
+        inputDate.y = inputDate.y.slice(2, 4);
+      } else if (toYearFormat > inputDate.yearFormat) {
+        const century = (+inputDate.y < 30) ? '20' : '19';
```

Comment:
optional, but you could compare without converting to number)

Reply:
But it's already a num)
![image](https://user-images.githubusercontent.com/99331793/172903488-7ccfe099-9ccf-433e-9d41-538164aeecb2.png)